File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/inheritance/overrideFromMultipleBaseClasses/twoInterfacesAsSuperClass/Base2.java
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.overrideFromMultipleBaseClasses.twoInterfacesAsSuperClass;

/**
 * Base 2, interface with default method, Second base class.
 *
 * @author Ranjith Krishnamurthy
 */
public interface Base2 {
    default public void print1() {
        System.out.println("Base2 --> print1");
    }

    default public void print2() {
        System.out.println("Base2 --> print2");
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/inheritance/overrideFromMultipleBaseClasses/twoInterfacesAsSuperClass/Base2.kt
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.overrideFromMultipleBaseClasses.twoInterfacesAsSuperClass

/**
 * Base 2, interface with default method, Second base class.
 *
 * @author Ranjith Krishnamurthy
 */
interface Base2 {
    fun print1() {
        println("Base2 --> print1")
    }

    fun print2() {
        println("Base2 --> print2")
    }
}